typing compile_fx.py

aorenste opened the pull request:
Type annotations for compile_fx.  
- Some of the stuff here is pretty complicated (functions which return functions that take functions) so I bailed on those and used `Any` just to get the rest landed.
- There are also changes to type signatures in other files which I did just to let mypy know more about the types in compile_fx.py.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #138033



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @rec

aorenste committed:
typing compile_fx.py

[ghstack-poisoned]

aorenste committed:
Update on "typing compile_fx.py"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang rec

[ghstack-poisoned]

aorenste committed:
Update on "typing compile_fx.py"

Type annotations for compile_fx.  
- Some of the stuff here is pretty complicated (functions which return functions that take functions) so I bailed on those and used `Any` just to get the rest landed.
- There are also changes to type signatures in other files which I did just to let mypy know more about the types in compile_fx.py.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang rec

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/codecache.py:
```diff
@@ -749,12 +751,12 @@ def __init__(
         self.fx_kwargs = {}
         for k in sorted(fx_kwargs):
```
```suggestion
        for k, v in sorted(fx_kwargs.items()):
```
May solve some of the type ignore literal errors.

aorenste committed:
Update on "typing compile_fx.py"

Type annotations for compile_fx.  
- Some of the stuff here is pretty complicated (functions which return functions that take functions) so I bailed on those and used `Any` just to get the rest landed.
- There are also changes to type signatures in other files which I did just to let mypy know more about the types in compile_fx.py.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang rec

[ghstack-poisoned]

Skylion007 commented on torch/_dynamo/repro/after_aot.py:
```diff
@@ -1,4 +1,7 @@
 # mypy: allow-untyped-defs
+
+from __future__ import annotations
```
I don't think you need this.

Skylion007 commented on torch/_inductor/__init__.py:
```diff
@@ -1,16 +1,22 @@
 # mypy: allow-untyped-defs
-from typing import Any, Dict, List, Optional, Tuple
+from __future__ import annotations
```
Definitely don't need this either

aorenste committed:
Update on "typing compile_fx.py"

Type annotations for compile_fx.  
- Some of the stuff here is pretty complicated (functions which return functions that take functions) so I bailed on those and used `Any` just to get the rest landed.
- There are also changes to type signatures in other files which I did just to let mypy know more about the types in compile_fx.py.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang rec

[ghstack-poisoned]

aorenste committed:
Update on "typing compile_fx.py"

Type annotations for compile_fx.  
- Some of the stuff here is pretty complicated (functions which return functions that take functions) so I bailed on those and used `Any` just to get the rest landed.
- There are also changes to type signatures in other files which I did just to let mypy know more about the types in compile_fx.py.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang rec

[ghstack-poisoned]

aorenste commented:
@pytorchbot merge
